src/base/static/components/input-form/index.js

```diff
@@ -254,14 +254,16 @@ class InputForm extends Component {
     this.setState({
       isFormSubmitting: true,
     });
-    const collection = this.props.places[this.selectedCategoryConfig.dataset];
+    const collection = this.props.places[
+      this.selectedCategoryConfig.dataset_slug
+    ];
     collection.add(
       {
         location_type: this.selectedCategoryConfig.category,
         datasetSlug: this.props.mapConfig.layers.find(
-          layer => this.selectedCategoryConfig.dataset === layer.id,
```

Comment:
There's a lot of confusion right now around the terminology used to describe datasets. It seems that the terms `datasetId`, `datasetSlug`, and `collectionId` are used interchangeably. I didn't refactor these inconsistencies in this PR, but I think we could clean this up easily in a future PR.

I was thinking we should assume that a dataset's slug is a unique identifier. The problem with using the actual numerical dataset id is that the id changes depending on which API instance we're building a flavor against.

So I'd propose that we use the terms `datasetSlug` and `dataset_slug` as unique identifiers on the frontend. What do you think?

Reply:
`datasetSlug` makes sense. Also, I thought we decided to start using camelCase for our config values (`dataset_slug` -> `datasetSlug`)?

Not a big deal either way.

Reply:
The convention for the config has been `snake_case`, though this is followed inconsistently. It's awkward, though, to see snake case properties in the codebase, so maybe we should convert to camel case.

Reply:
I would favor camelCase for all new config names, since that's the convention for js. Snake case was probably left over from the original implementation, which was done in Python.